Package.swift

```diff
@@ -24,12 +24,14 @@ let package = Package(
     ],
     dependencies: [
         .package(url: "https://github.com/WeTransfer/Mocker", from: "3.0.1"),
+        .package(url: "https://github.com/iwill/generic-json-swift", from: "2.0.2"),
```

Comment:
Hey @eligutovsky , have we had any discussion about adding new dependencies to the SDK?

Some clients are a little sensitive to the size/dependency footprint, and dependencies can make integration more tricky if there are conflicting versions.

Just a thought, we should consider it carefully.

Reply:
This dependency was copy-pasted to the source code previously. I will remove this dependency and will keep it copy-pasted.

Reply:
Sorry not sure I understand, you mean we already depend on it? (if so, fair enough, might propose the inverse of dependency simplification in the future, but that's another topic 😅)

Reply:
Yes, we're already depend on it. See `OptimoveSDK/Sources/Classes/JSON/JSON.swift` -- in removed files of this PR.

Reply:
Ah, clear!

Next question: do you know why we 'vendored' this library code into the SDK in the past, does it matter?

Reply:
It was added previously because the SDK allows to get `[String: Any]` as a parameter of an event. 

See https://github.com/optimove-tech/Optimove-SDK-iOS/pull/97#discussion_r1426850648

It might worth to replace `JSON` with `Data` by using `JSONSerialization`.

Reply:
Ideally, we need to decode any internal json to a Codable model.